Tests/CommandsTests/SwiftToolTests.swift

```diff
@@ -31,6 +31,8 @@ final class SwiftToolTests: CommandsTestCase {
                 tool.observabilityScope.emit(info: "info")
                 tool.observabilityScope.emit(debug: "debug")
 
+                tool.waitForObservabilityEvents(timeout: .now() + .seconds(1))
```

Comment:
Is this requirement new or was this always a flakiness in the tests?  It makes sense conceptually to wait for the output, just curious.

Reply:
in this specific case, there is a new queue we are introducing so the tests would be flaky without a wait.

more broadly, I believe we were always subject to loosing the last messages, and now that we have queue sync we can wait on we have the opportunity to wait on it